boards/posix/native_sim/doc/index.rst

```diff
@@ -3,18 +3,33 @@
 Native simulator - native_sim
 #############################
 
+.. contents::
+   :depth: 1
+   :backlinks: entry
+   :local:
```

Comment:
just to confirm, in case this was unintentional, that you don't want C Library Choice under How to use it?

Reply:
> just to confirm, in case this was unintentional, that you don't want C Library Choice under How to use it?

Not really. 
The 32/64 did not really feel like it needed to be a subsection either, but it seemed ok either way to me.